Baikal/PostEffects/ML/denoiser_preprocess.cpp

```diff
@@ -0,0 +1,252 @@
+                                               CL_MEM_READ_WRITE,
+                                               4 * width * height);
+
+            m_input = CLWBuffer<float>::Create(context,
+                                               CL_MEM_READ_WRITE,
+                                               m_channels * width * height);
+
+            ml_image_info image_info = {ML_FLOAT32, m_width, m_height, m_channels};
+            m_image = mlCreateImage(m_context.get(), &image_info);
+
+            if (!m_image)
+            {
+                throw std::runtime_error("can not create ml_image");
+            }
+
+            m_is_initialized = true;
+        }
+
+        Image DenoiserPreprocess::MakeInput(PostEffect::InputSet const& inputs)
+        {
+            auto context = GetContext();
+            unsigned channels_count = 0u;
+            float real_sample_count = 0.f;
+
+            if (!m_is_initialized)
+            {
+                auto color = inputs.at(Renderer::OutputType::kColor);
+                m_width = color->width();
+                m_height = color->height();
+                Init(m_width, m_height);
+                m_is_initialized = true;
+            }
+
+            for (const auto& desc : m_layout)
+            {
+                auto type = desc.first;
+                auto input = inputs.at(type);
+
+                auto clw_output = dynamic_cast<ClwOutput*>(input);
+                auto device_mem = clw_output->data();
+
+                switch (type)
+                {
+                    case OutputType::kColor:
+                    {
+                        DivideBySampleCount(CLWBuffer<float3>::CreateFromClBuffer(m_cache),
+                                            device_mem);
+
+                        WriteToInputs(m_input,
+                                      m_cache,
+                                      m_width,
+                                      m_height,
+                                      channels_count, // dst channels offset
+                                      m_channels,     // dst channels num
+                                      0,              // src channels offset
+                                      4,              // src channels num
+                                      3);             // channels to copy
+
+                        channels_count += 3;
+
+                        context.ReadBuffer<float>(0, m_cache, &real_sample_count, 3, 1).Wait();
+
+                        if (static_cast<std::uint32_t>(real_sample_count) < m_start_spp)
+                        {
+                            return Image(static_cast<std::uint32_t>(real_sample_count), nullptr);
+                        }
+                        break;
+                    }
+                    case OutputType::kDepth:
+                    {
+                        auto normalized_buf = CLWBuffer<cl_float3>::CreateFromClBuffer(m_cache);
```

Comment:
This variable may be eliminated: it's used actually once in Normalize, the 2nd usage is a "cast", it can be done from m_cache directly.
When the variable is eliminated, the surrounding curly brackets can be removed, as well as brackets for other `case`s.